### PR TITLE
Ruby Group Capstone - Catalog of My Things

### PR DESCRIPTION
**Pull request by PraisesPJMT**
# Ruby Group Capstone - Catalog of My Things
## Project Link
- Project Kanban Link: [https://github.com/users/PraisesPJMT/projects/2](https://github.com/users/PraisesPJMT/projects/2)
- Repository Link: [https://github.com/PraisesPJMT/Personal-Catalog](https://github.com/PraisesPJMT/Personal-Catalog)

## Team Members
Here are the team members working on this project.
- Member 1: Benard Pacho (@benardop)
- Member 2: Praises Tula (@PraisesPJMT)
- Member 3: Shaquille Ndunda (@shaqdeff)

## Feature Description
- Created the following classes in line with the [UML class diagram](https://github.com/microverseinc/curriculum-ruby/blob/main/group-capstone/images/catalog_of_my_things.png).
  - `Item`, `Game`, `Book`, `MusicAlbum`, `Genre`, `Label`, and `Author`
- Wrote unit tests for all the classes listed above
- Create a `main.rb` file that will serve as your console app entry-point.
- When initiated, the console app:
  - Present the user with a list of options to perform.
  - Let users choose an option.
  - Input for parameters for corresponding options.
  - Quit the app with the option number `10`.
- Preserve all session data as local storage in JSON format.

Happy Reviewing!!!!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,151 @@
...
+### Install
+Install this project with:
+```sh
+  gem install bundler
+  bundle install
+```
+### Usage
+To run the project, execute the following command:
+```sh
+  ruby solver.rb
```
Comment (brytebee) [inline README.md]: - Kindly update this to `ruby main.rb`. This command is misleading and throws an error. This makes your README professional.

image

File: spec/author_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+require_relative '../helper'
+
+describe Author do
+  before :each do
+    @author = Author.new 'Jimmy', 'Cruxland'
+  end
+
+  describe '#new' do
+    it 'returns a new instance of Author object' do
+      @author.should be_an_instance_of Author
```
Comment (brytebee) [inline spec/author_spec.rb]: - [x] **Optional:** The **should expectation** used throughout your tests is deprecated and can be replaced by the **`expect` block** as below:

```
expect(@author).to be_an_instance_of Author
```

File: spec/genre_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    it 'Name variable of Genre instance should be mutable' do
+      test_genre.name = 'Soul'
+      expect(test_genre.name).to eq 'Soul'
+      test_genre.name = 'Gospel'
+      expect(test_genre.name).to eq 'Gospel'
+    end
+
+    it 'Changing @id variable of Genre instance should raise and NoMethodError error' do
+      expect { test_genre.id = 12 }.to raise_error
+      expect { test_genre.id = 678 }.to raise_error
```
Comment (brytebee) [inline spec/genre_spec.rb]: - [x] **Required:** This error prevents the tests from running correctly. Kindly read the error message in the picture and fix it.

image

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @PraisesPJMT & @benardop & @shaqdeff 👋, 

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great UI design.✔️
- App functions as it should.✔️
- Great use of Ruby style guide and coding style.✔️
- Clean code.✔️

##

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (PraisesPJMT) [reply README.md]: Noted! We just updated that to `ruby main.rb`

File: spec/author_spec.rb (hunk 2 above)
Comment (PraisesPJMT) [reply spec/author_spec.rb]: Noted! We just updated the tests with modern semantics.

File: spec/genre_spec.rb (hunk 3 above)
Comment (PraisesPJMT) [reply spec/genre_spec.rb]: Noted! We effected the correction by matching the expected error type.

**Comment by PraisesPJMT** [comment]
## Review
Thanks so much @brytebee for the feedback!
## Changes Made
The project team made the following changes in line with the review recommendations
- Update `README.md` file to reflect project progress
- Replaced deprecated semantics used in test files with modern semantics
- Match error types for all taste cases involving error checks

Happy Reviewing!!!!!

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi team, 
## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
README is professional. :heavy_check_mark:
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by PraisesPJMT** [comment]
Thanks so much @SajjadAhmad14 for the amazing review!

Happy Weekend!!!!!!!!!!!!!!!!

**Pull request merged**